dluhc-component-library/src/stories/DatasetSelection.stories.tsx

```diff
@@ -0,0 +1,114 @@
+import {
+  QueryClient,
+  QueryClientProvider,
+  useQuery,
+} from "@tanstack/react-query";
+import { fromExtent } from "ol/geom/Polygon";
+import { useState } from "preact/compat";
+import { ReactNode } from "react";
+import { fetchDatasetList } from "src/api/api";
+import { Dataset } from "src/api/types";
+import DatasetList from "src/components/datasets/DatasetList";
+import DatasetLayer from "src/components/maps/DatasetLayer";
+import BaseMap from "src/components/maps/baseMap";
+import MapContainer from "src/components/maps/mapContainer";
+import { useMap } from "src/contexts/mapContext";
+
+const queryClient = new QueryClient();
+
+interface MapComponentProps {
+  lat: number;
+  lng: number;
+  zoom: number;
+  selectedDatasets: Dataset[];
+}
+
+interface DatasetSelectionProps {
+  lat: number;
+  lng: number;
+  zoom: number;
+}
+
+const MapComponent = ({
+  lat,
+  lng,
+  zoom,
+  selectedDatasets,
+}: MapComponentProps) => {
+  const map = useMap();
+  const area = fromExtent(map.getView().calculateExtent(map.getSize()));
+  return (
+    <>
+      <BaseMap lat={lat} lng={lng} zoom={zoom} className="h-full" />
+      {selectedDatasets.map((dataset) => (
+        <DatasetLayer dataset={dataset} area={area} key={dataset.dataset} />
+      ))}
+    </>
+  );
+};
+
+const DatasetSelection = ({ lat, lng, zoom }: DatasetSelectionProps) => {
+  const { data: datasets, isLoading } = useQuery({
+    queryKey: ["datasets"],
+    queryFn: () => fetchDatasetList(),
+    select: (data) =>
+      data.datasets.sort((a, b) => a.name.localeCompare(b.name)),
+  });
+
+  const [selectedDatasets, setSelectedDatasets] = useState<Dataset[]>([]);
+
+  const onSelect = (dataset: Dataset) => {
```

Comment:
`Non-blocker` Not a big fan of the way we are managing data here. I would expect the onSelect to only be working with the "names" of the data sets rather than comparing full datasets against each other.